source/common/http/codec_client.cc

```diff
@@ -117,6 +117,9 @@ void CodecClient::onEvent(Network::ConnectionEvent event) {
               StreamInfo::ResponseFlag::UpstreamProtocolError);
         }
       }
+    } else {
+      ENVOY_CONN_LOG(warn, "Connection is closed by {} during connecting.", *connection_,
```

Comment:
Curious. How do we get a Local/Remote close event but connected_ is still true (in the previous branch)?

Reply:
connected_ flips to true after Connect event is raised which for a QUIC connection is after handshake and never gets changed during disconnecting.  There is member variable to remember disconnected state in CodecClient, but I remember there is some in its subclasses which flips upon onEvent()